### PR TITLE
Allow individual tests override wait timeouts and retry intervals 

### PR DESCRIPTION
**Pull request by alexeykazakov**
For example if we need to override the default timeout:
```
s.hostAwait.WithRetryOptions(TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(...)
```

It's useful in test which wait for some resource not to be created and we don't want to wait for the entire default 120 seconds in such tests.

### CODE REVIEW

File: wait/awaitility.go [hunk 1]
```diff
@@ -54,10 +60,27 @@ type SingleAwaitility interface {
...
-	T               *testing.T
-	Client          framework.FrameworkClient
-	Ns              string
-	OtherOperatorNs string
+	T                     *testing.T
+	Client                framework.FrameworkClient
+	Ns                    string
+	OtherOperatorNs       string
+	OperatorRetryInterval time.Duration
+	OperatorTimeout       time.Duration
```
Comment (MatousJobanek) [inline wait/awaitility.go]: why do we have `OperatorRetryInterval` and `OperatorTimeout` when they are not used anywhere?

Comment (alexeykazakov) [reply wait/awaitility.go]: Good question :)
Let me double check and clean them up.

Comment (alexeykazakov) [reply wait/awaitility.go]: Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/46/commits/44cc75736a74e317729701874ce1ed520d54208a

**Comment by alexeykazakov** [comment]
I also did some cleanup. Removed the garbage collection test from UserSignup tests because it's already covered in the overall e2e_test.go.

**Pull request merged**